data/src/main/java/me/androidbox/data/worker_manager/TaskCreateSynchronizerWorker.kt

```diff
@@ -0,0 +1,78 @@
+package me.androidbox.data.worker_manager
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import me.androidbox.data.local.dao.TaskDao
+import me.androidbox.data.local.entity.TaskEntity
+import me.androidbox.data.mapper.toTaskDto
+import me.androidbox.data.remote.network.task.TaskService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.data.worker_manager.constant.Constant.RETRY_COUNT
+import me.androidbox.domain.constant.SyncAgendaType
+
+@HiltWorker
+class TaskCreateSynchronizerWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val taskService: TaskService,
+    private val taskDao: TaskDao
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
+        val createdTasks: Deferred<List<TaskEntity>>
+
+        /** Get all the created tasks */
+        supervisorScope {
+            createdTasks = async {
+                taskDao.getAllTasksBySyncType(SyncAgendaType.CREATE).map {
+                    it.task
+                }
+            }
+        }
```

Comment:
The `supervisorScope` and `async` block doesn't do anything here, since the coroutine will suspend until the tasks are retrieved anyways. You can just do `val createdTasks = taskDao.getAllTasksBySyncType(...)` as you're not executing anything in parallel here

Reply:
Exactly what philipp mentions

Sinxe you are just making 1 call to the repository, and not multiple, ir makes no sense to have a supervisorscope and an async